jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -63,16 +64,54 @@ impl Block {
     }
 
     /// The transactions contained in the block
-    pub fn transactions(&self, context: &Context) -> FieldResult<Vec<Transaction>> {
-        Ok(self
-            .get_explorer_block(&context.db)?
-            .transactions
-            .iter()
-            .map(|(id, _tx)| Transaction {
-                id: id.clone(),
-                in_block: self.hash.clone(),
-            })
-            .collect())
+    pub fn transactions(
+        &self,
+        first: Option<i32>,
+        last: Option<i32>,
+        before: Option<IndexCursor>,
+        after: Option<IndexCursor>,
+        context: &Context,
+    ) -> FieldResult<TransactionConnection> {
+        let explorer_block = self.get_explorer_block(&context.db)?;
+        let mut transactions: Vec<&ExplorerTransaction> =
+            explorer_block.transactions.values().collect();
+
+        // TODO: This may be expensive at some point, but I can't rely in
+        // the HashMap's order (also, I'm assuming the order in the block matters)
```

Comment:
There's `LinkedHashMap` if that's of any help, it is already in use in `network`.